backend/models/helper/model.js

```diff
@@ -225,15 +225,18 @@ function importFail(account, model, sharedSpacePath, user, errCode, errMsg) {
  * @param {model} model - Model
  * @param {user} user - The user who triggered the status
  */
-function setStatus(account, model, status, user) {
-	ChatEvent.modelStatusChanged(null, account, model, { status, user });
-	return ModelSetting.findById({account, model}, model).then(setting => {
-		setting.status = status;
+async function setStatus(account, model, status, user) {
+	let setting;
```

Comment:
@carmenfan Linter does not like the suggested change; `setting` within `try` is considered unused because the return is outside scope of the try and return thinks `setting` is undefined.

Reply:
you shouldn't need to return setting on L239. 
setting is undefined so you're just really returning undefined, which (if linter allows) means you can just move L239 within the try bracket